pydoctor/epydoc/markup/plaintext.py

```diff
@@ -34,10 +35,20 @@ class ParsedPlaintextDocstring(ParsedDocstring):
     def __init__(self, text: str):
         ParsedDocstring.__init__(self, ())
         self._text = text
+         # Caching:
+        self._document: Optional[nodes.document] = None
 
     @property
     def has_body(self) -> bool:
         return bool(self._text)
 
     def to_stan(self, docstring_linker: DocstringLinker) -> Tag:
         return tags.p(self._text, class_='pre')
+    
+    def to_node(self) -> nodes.document:
+        if self._document is not None:
+            return self._document
+        else:
+            self._document = utils.new_document('plaintext')
+            self._document.children.append(nodes.literal_block(rawsource=self._text, text=self._text))
+            return self._document
```

Comment:
This code is currently unused, maybe it should raise NotImplementedError() instead ?

Reply:
I'm not sure what you mean, the `to_stan()` default implementation raises NotImplementedError, so all subclasses of ParsedDocstring MUST override `to_stan()`.

I agree with trying to keep markup modules deal only with nodes tree, but it's not the way it has been working for years. Please be more specific regarding what changes you what to see in this PR, what changes you want to see in another, future, PR.

Reply:
OK. 

I've implemented a middle ground approach where `ParsedDocstring` provides a default implementation for `to_stan()`, this implementation is really simple and relies on code in `node2stan` module.

Reply:
That's a step in the right direction.

However, what I don't understand is why `ParsedPlaintextDocstring` needs to override `to_stan()`. A comment says "Also, the consolidated fields handling in restructuredtext.py relies on this "pre" class.", but I can't find that back in the code. I do see `ParsedPlaintextDocstring` being used as a fallback for field splitting, but I don't see any dependency on Stan there. Maybe I'm just overlooking it though.

If at some point `to_stan()` is not overridden by any subclass, it could be moved out of `ParsedDocstring`.

Note that the epydoc code was only imported into pydoctor as part of the Python 3 porting in 2019. A few months later I reduced the epytext -> HTML -> Stan -> HTML pipeline to just epytext -> Stan -> HTML, which is when the Stan dependency was added to the epytext code. So the current situation hasn't existed for all that long.

Stan isn't very good for introspection, both because its API isn't designed for that and because the generated HTML is not semantic enough. Therefore implementing things like summary auto-generation on Stan trees would be very error prone. I hope it will be easier to do on docutils trees.

Reply:
Happy that you like the changes. 

Regarding why ParsedPlaintextDocstring needs to override to_stan(). Please look at this [test case](https://github.com/twisted/pydoctor/blob/master/pydoctor/test/epydoc/restructuredtext.doctest#L43). My opinion it that it touches stuff outside the scope of this PR, so I'de rather not modify this behaviour now. 

> I hope it will be easier to do on docutils trees.

Yes it is, I think.

Reply:
The new "pipeline" after this PR will be: 

Epytext -> Docutils -> Stan (with `HTMLTranslator` and `html2stan()`) -> HTML